packages/client/src/v2-events/features/events/AdvancedSearch/SearchModifier.tsx

```diff
@@ -38,21 +39,55 @@ const SearchParamContainer = styled.div`
   }
 `
 
+function convertPathToLabel(path: string): string {
+  if (!path.includes('.')) {
+    return path.charAt(0).toUpperCase() + path.slice(1)
+  }
+
+  const parts = path.split('.')
+  const capitalizedFirst = parts[0].charAt(0).toUpperCase() + parts[0].slice(1)
+  return [capitalizedFirst, ...parts.slice(1)].join(' ')
+}
+
 export function SearchModifierComponent({
+  eventType,
   searchParams
 }: {
+  eventType: string
   searchParams: Record<string, string>
 }) {
   const navigate = useNavigate()
   const intl = useIntl()
+  const searchParamsAsLabel = Object.entries(searchParams).map(
+    ([key, value]) => ({ key: convertPathToLabel(key), value })
+  )
 
   return (
     <>
       <SearchParamContainer>
+        {[
+          {
+            key: intl.formatMessage(constantsMessages.event),
+            value: convertPathToLabel(eventType)
+          },
+          ...searchParamsAsLabel
+        ].map((pill, index) => {
+          const label = `${pill.key} : ${pill.value}`
+          return (
+            <Pill
+              key={`${label}${index}`}
+              label={label}
+              size="small"
+              type="default"
+            ></Pill>
+          )
+        })}
```

Comment:
Neater and more readable version:

```suggestion
  const eventSearchParamLabel = `${intl.formatMessage(constantsMessages.event)} : ${convertPathToLabel(eventType)}`
  const searchParamLabels = Object.entries(searchParams).map(
    ([key, value]) => `${convertPathToLabel(key)} : ${value}`
  )

  const allSearchParamLabels = [eventSearchParamLabel, ...searchParamLabels]

  return (
    <>
      <SearchParamContainer>
        {allSearchParamLabels.map((label) => (
          <Pill key={label} label={label} size="small" type="default"></Pill>
        ))}
```